### PR TITLE
Fixes for code review comments from https://github.com/Azure/azure-rest-api-specs/pull/752

### PR DESCRIPTION
**Pull request by naveedaz**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [x] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

### CODE REVIEW

File: arm-web/2016-03-01/DeletedWebApps.json [hunk 1]
```diff
@@ -415,24 +404,23 @@
...
               "type": "integer"
             },
             "suspendedTill": {
-              "format": "date-time",
               "description": "App suspended till in case memory-time quota is exceeded.",
               "type": "string",
               "readOnly": true
             },
+            "advancedRoutingEnabled": {
+              "type": "boolean"
```
Comment (amarzavery) [inline arm-web/2016-03-01/DeletedWebApps.json]: @naveedaz - missing description.

**Review by jianghaolu — CHANGES_REQUESTED** [review]

**Comment by fearthecowboy** [comment]
>       "pattern": "^[-\\\\w\\\\._\\\\(\\\\)]+[^\\\\.]$"

We believe this is over-escaped. 

The intent would be to be left with 
     ^[-\w\._\(\)]+[^\.]$

which means the literal in JSON should be

   ^[-\\w\\._\\(\\)]+[^\\.]$



---
Refers to: arm-web/2016-03-01/DeletedWebApps.json:1078 in c26c550. [](commit_id = c26c550c38f79918433a2aadd13ae59f6ed52e7f, deletion_comment = False)

**Comment by fearthecowboy** [comment]
>       "description": "Custom action to be executed\r\n            when an auto heal rule is triggered.",

Still seeing some \r in descriptions.

---
Refers to: arm-web/2016-03-01/DeletedWebApps.json:121 in c26c550. [](commit_id = c26c550c38f79918433a2aadd13ae59f6ed52e7f, deletion_comment = False)

**Comment by fearthecowboy** [comment]
>       "description": "Custom action to be executed\r\n            when an auto heal rule is triggered.",

and collapse extra spaces to single space.

---
Refers to: arm-web/2016-03-01/DeletedWebApps.json:121 in c26c550. [](commit_id = c26c550c38f79918433a2aadd13ae59f6ed52e7f, deletion_comment = False)

File: arm-web/2016-03-01/DeletedWebApps.json [hunk 2]
```diff
@@ -162,7 +154,6 @@
           "description": "A rule based on total requests."
         },
         "privateBytesInKB": {
-          "format": "int32",
```
Comment (fearthecowboy) [inline arm-web/2016-03-01/DeletedWebApps.json]: >"format": "int32", [](start = 10, length = 18)

Why did this get removed? This should still be there.

File: arm-web/2016-03-01/DeletedWebApps.json [hunk 3]
```diff
@@ -257,7 +248,6 @@
           "description": "ARM envelope properties",
           "properties": {
             "deletedTimestamp": {
-              "format": "date-time",
```
Comment (fearthecowboy) [inline arm-web/2016-03-01/DeletedWebApps.json]: >"format": "date-time", [](start = 14, length = 22)

This shouldn't have been removed. Generation problem?

File: arm-web/2016-03-01/DeletedWebApps.json [hunk 4]
```diff
@@ -257,7 +248,6 @@
...
               "description": "Time when the app was deleted.",
```
Comment (fearthecowboy) [inline arm-web/2016-03-01/DeletedWebApps.json]: >"Time when the app was delete [](start = 29, length = 29)

Time zone? UTC?

**Comment by fearthecowboy** [comment]
>               "description": "Current state of the app. Read-only.",

Saying this in the description is not advised. In Some languages this isn't even correct in the generated code. 

---
Refers to: arm-web/2016-03-01/DeletedWebApps.json:256 in c26c550. [](commit_id = c26c550c38f79918433a2aadd13ae59f6ed52e7f, deletion_comment = False)

**Comment by fearthecowboy** [comment]
>               "description": "Hostnames associated with the app. Read-only.",

See above.


---
Refers to: arm-web/2016-03-01/DeletedWebApps.json:261 in c26c550. [](commit_id = c26c550c38f79918433a2aadd13ae59f6ed52e7f, deletion_comment = False)

**Comment by fearthecowboy** [comment]
>               "description": "Name of the repository site. Read-only.",

See above. (and below!) :D

---
Refers to: arm-web/2016-03-01/DeletedWebApps.json:269 in c26c550. [](commit_id = c26c550c38f79918433a2aadd13ae59f6ed52e7f, deletion_comment = False)

File: arm-web/2016-03-01/DeletedWebApps.json [hunk 5]
```diff
@@ -340,7 +330,6 @@
               ]
             },
             "lastModifiedTimeUtc": {
-              "format": "date-time",
```
Comment (fearthecowboy) [inline arm-web/2016-03-01/DeletedWebApps.json]: >"date-time" [](start = 24, length = 11)

Lost format again. :(

**Comment by fearthecowboy** [comment]
>           "description": "Gets or sets a JSON string containing a list of dynamic tags that will be evaluated from user claims in the push registration endpoint.",

Please remove "Gets or sets" from descriptions - the language generator will make appropriate text regarding property accessors.


---
Refers to: arm-web/2016-03-01/DeletedWebApps.json:602 in c26c550. [](commit_id = c26c550c38f79918433a2aadd13ae59f6ed52e7f, deletion_comment = False)

**Review by fearthecowboy — CHANGES_REQUESTED** [review]
Hey @naveedaz We've done a cursory run over a couple files; we're seeing a few issues with missing `format` and a couple other things. Could you fix them up and re-generate and push a new set of files? 

Thanks!

**Review by jianghaolu — APPROVED** [review]

**Comment by yugangw-msft** [comment]
Please port the support to create linux plan, particularly [this commit](https://github.com/Azure/azure-rest-api-specs/commit/90c584a99311109bdbfc2d15ad2b2fb70c0896e9)
/cc: @LukaszStem

**Comment by yugangw-msft** [comment]
Also i still saw we have property names using `serverFarm`, like [here](https://github.com/Azure/azure-rest-api-specs/blob/66457e2770830e2c9d3dc257e0c57d31dbf0f483/arm-web/2016-09-01/AppServicePlans.json#L2418)

**Comment by naveedaz** [comment]
@yugangw-msft Changing the property serverFarmId will break a lot of things. The resources in ARM themselves are serverfarms. Is this really needed right now?

**Comment by yugangw-msft** [comment]
>@yugangw-msft Changing the property serverFarmId will break a lot of things. The resources in ARM themselves are serverfarms. Is this really needed right now?

@naveedaz Not critical, you can do it later

**Comment by amarzavery** [comment]
@naveedaz - We ran the semantic validator and found some issues. Can you please resolve them

- TopLevelDomains.json has a global parameter named "resourceGroupNameParameter" but it is not being used anywhere. Can you please remove it?
> Semantically validating  https://raw.githubusercontent.com/naveedaz/azure-rest-api-specs/master/arm-web/2015-04-01/TopLevelDomains.json:


Warnings
--------
[ { code: 'UNUSED_PARAMETER',
    message: 'Parameter is defined but is not used: #/parameters/resourceGroupNameParameter',
    path: [ 'parameters', 'resourceGroupNameParameter' ] } ]

- Recommendations.json has an unused model definition "Resource". Can you please remove it from that spec?
> Semantically validating  https://raw.githubusercontent.com/naveedaz/azure-rest-api-specs/master/arm-web/2016-03-01/Recommendations.json:


Warnings
--------
[ { code: 'UNUSED_DEFINITION',
    message: 'Definition is defined but is not used: #/definitions/Resource',
    path: [ 'definitions', 'Resource' ] } ]

**Comment by fearthecowboy** [comment]
@naveedaz if you can correct the issue brought up by @amarzavery ,  we're prepared to merge this PR; and then we can move you on to creating SDK and Unit Tests.

File: arm-web/2016-08-01/WebApps.json [hunk 6]
```diff
@@ -10324,9 +9947,6 @@
             "$ref": "#/parameters/subscriptionIdParameter"
           },
           {
-            "$ref": "#/parameters/resourceGroupNameParameter"
-          },
-          {
```
Comment (jianghaolu) [inline arm-web/2016-08-01/WebApps.json]: Server returns 200 as well. Same for `stop()`, `restart()`, `startSlot()`, `restartSot()`, `stopSlot()`.

**Review by jianghaolu — CHANGES_REQUESTED** [review]
https://github.com/Azure/azure-rest-api-specs/blame/master/arm-web/2016-08-01/WebApps.json#L13491
`scmType` should be writable.

**Review by fearthecowboy — APPROVED** [review]

**Comment by fearthecowboy** [comment]
@jianghaolu Are the changes you requested completed?

**Comment by jianghaolu** [comment]
No. The 2 fixes I'm still waiting are:
1) `scmType` should be read-write
2) `start()`, `stop()`, `restart()`, `startSlot()`, `restartSot()`, `stopSlot()` should accept 200

**Comment by jianghaolu** [comment]
Another fix required:
`CertitificateDetails` shouldn't have an extra layer of `properties`.

**Comment by naveedaz** [comment]
I have made the changes requested by @jianghaolu. Thanks

**Comment by jianghaolu** [comment]
Thanks Navy, do you know about why `CertificateDetails`'s definition here is different from the actual payload I received from server?

**Comment by jianghaolu** [comment]
Looks like the issue for `CertificateDetails` is fixed.

**Review by jianghaolu — APPROVED** [review]

**Comment by naveedaz** [comment]
Can we get this merged please?

**Pull request merged**